chainindex/gc.go

```diff
@@ -0,0 +1,113 @@
+package chainindex
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/filecoin-project/lotus/chain/actors/policy"
+)
+
+var (
+	log             = logging.Logger("chainindex")
+	cleanupInterval = time.Duration(8) * time.Hour
+)
+
+func (si *SqliteIndexer) gcLoop() {
+	defer si.wg.Done()
+
+	// Initial cleanup before entering the loop
+	si.cleanupRevertedTipsets(si.ctx)
+	si.gc(si.ctx)
+
+	cleanupTicker := time.NewTicker(cleanupInterval)
+	defer cleanupTicker.Stop()
+
+	for {
+		select {
+		case <-cleanupTicker.C:
+			si.cleanupRevertedTipsets(si.ctx)
+			si.gc(si.ctx)
+		case <-si.ctx.Done():
+			return
+		}
+	}
+}
+
+func (si *SqliteIndexer) gc(ctx context.Context) {
+	if si.gcRetentionEpochs <= 0 {
+		return
+	}
+
+	head := si.cs.GetHeaviestTipSet().Height()
+	removeEpoch := int64(head) - si.gcRetentionEpochs
```

Comment:
```suggestion
	head := si.cs.GetHeaviestTipSet()
	removeEpoch := int64(head.Height()) - si.gcRetentionEpochs
```

sorry, not incorrect, we just have such a strong convention across the codebase of "head" being the tipset, not the height, so to minimise any potential for confusion